File: projetoReadyQueue/roundRobin.js
const roundRobin = (processes, quantum) => {
    let ms = 0
    while (processes.length > 0) {
        processes.forEach(process => {
            for(let i = 0; i < quantum; i++) {
                if (process.burstTime > 0) {
                    console.log(`Processo: ${process.name}: ${process.burstTime}`)
                    ms++
                    console.log(`Tempo: ${ms}`)
                    process.burstTime--
                }
            }
            if (process.burstTime <= 0) {
                processes.splice(processes.indexOf(process), 1)
            }
        });
    }
}

p1 = { name: "p1", burstTime: 24 }

p2 = { name: "p2", burstTime: 3 }

p3 = { name: "p3", burstTime: 3 }

processes = [p1, p2, p3]

roundRobin(processes, 4)

File: projetoReadyQueue/roundRobin.py
import time

class Process:
    def __init__(self, name, burstTime):
        self.name = name
        self.burstTime = burstTime

def calculate_total_burst_time(processes):
    total_burst_time = 0
    for process in processes:
        total_burst_time += process.burstTime
    return total_burst_time

def roundRobin(processes, quantum):
    ms = 0
    total_burst_time = calculate_total_burst_time(processes)
    
    while total_burst_time > 0:
        for process in processes:
            for _ in range(quantum):
                if process.burstTime > 0:
                    print("Processo: " + process.name + " - BurstTime: " + str(process.burstTime))
                    ms += 1
                    print("Tempo: " + str(ms))
                    process.burstTime -= 1
                    total_burst_time -= 1
                    time.sleep(1)

p1 = Process("p1", 24)
p2 = Process("p2", 3)
p3 = Process("p3", 3)

processes = [p1, p2, p3]
roundRobin(processes, 4)



